rasa/nlu/selectors/response_selector.py

```diff
@@ -450,10 +450,10 @@ def process(self, message: Message, **kwargs: Any) -> None:
         prediction_dict = {
             RESPONSE_SELECTOR_PREDICTION_KEY: {
                 "id": top_label["id"],
-                RESPONSE_SELECTOR_RESPONSES_KEY: label_response_templates,
+                RESPONSE_SELECTOR_RESPONSES_KEY: label_responses,
                 PREDICTED_CONFIDENCE_KEY: top_label[PREDICTED_CONFIDENCE_KEY],
                 INTENT_RESPONSE_KEY: label_intent_response_key,
-                RESPONSE_SELECTOR_TEMPLATE_NAME_KEY: util.intent_response_key_to_template_key(
+                RESPONSE_SELECTOR_UTTER_ACTION_KEY: util.intent_response_key_to_template_key(
```

Comment:
To preserve compatibility with the old format we'll have to duplicate the result across the old and new key just as how we do [here](https://github.com/RasaHQ/rasa/pull/6964/files#diff-4d618e8bd1a6298b12f14a7bb6b7489a1de6b937e875519139422d9049674cedR39-R42) I believe. But not sure if duplication is the best thing to do. Opinions?

Reply:
This key is a bit tricky, as it's used all across `rasa/core` and `rasa/nlu`. We could try to duplicate this on the nlu side, and just keep the `utter_action` terminology on the `rasa/core` side. Let me experiment a bit now.

Reply:
We can also rename the constant, while keeping the value of the constant the same?

Reply:
> We can also rename the constant, while keeping the value of the constant the same?

Yes, that would be better but then it may be defeats the purpose of the PR for this component? Don't have strong opinions either ways.

Reply:
How do you feel about this format?
```
{
   "response":{
      "id":322111579505178260,
      "responses":[
         {
            "text":"I am Mr. Bot"
         }
      ],
      "confidence":0.5000246167182922,
      "intent_response_key":"chitchat/ask_name",
      "utter_action":"utter_chitchat/ask_name",
      "template_name":"utter_chitchat/ask_name"
   },
   "ranking":[
      {
         "id":322111579505178260,
         "confidence":0.5000246167182922,
         "intent_response_key":"chitchat/ask_name"
      },
      {
         "id":2114865043041600294,
         "confidence":0.49997538328170776,
         "intent_response_key":"chitchat/ask_weather"
      }
   ]
}
```

Reply:
I did the changes to get the format above, please let me know if this works out. 🙏🏼

Reply:
I'll re-run the tests too.

Reply:
This looks good. Do we want to duplicate `response_templates` as well by adding the `responses` key as well with the same values. Then we can add a warning that `response_templates` will also be deprecated and `responses` should be used?

Reply:
Yeah good point, if we're doing it with `utter_action`, might as well do that for `responses` too.